### PR TITLE
feat: refactor PIA and CAA getters

### PR DESCRIPTION
**Pull request by shuffledex**

### CODE REVIEW

File: src/api/procedures/controllerTransfer.ts
Comment (monitz87) [inline src/api/procedures/controllerTransfer.ts:91]: We can replace this with the current DID

File: src/api/procedures/launchSto.ts
Comment (monitz87) [inline src/api/procedures/launchSto.ts:247]: After a talk with @VictorVicente , we decided to go ahead and make the breaking change of requiring the user to pass `offeringPortfolio` instead of making it optional (please change the interface comments to reflect this)

This means that it's no longer required to fetch the PIAs here

File: src/api/procedures/redeemToken.ts
Comment (monitz87) [inline src/api/procedures/redeemToken.ts:44]: Since `getAuthorizations` checks that the current identity is a PIA, we can just use the current DID here

File: src/api/procedures/__tests__/redeemToken.ts
Comment (monitz87) [inline src/api/procedures/__tests__/redeemToken.ts:204]: Wrong description

**Review by monitz87 — CHANGES_REQUESTED** [review]

File: src/api/procedures/launchSto.ts
Comment (monitz87) [reply src/api/procedures/launchSto.ts:247]: Remember to commit it as a breaking change

File: src/api/entities/SecurityToken/CorporateActions/__tests__/index.ts
Comment (shuffledex) [reply src/api/entities/SecurityToken/CorporateActions/__tests__/index.ts:184]: The AgentGroup is wrapped in an Option type. I have to cover it anyway. I will ask core team if it is possible or not. In that case, they will need to change the schema.

File: src/api/entities/SecurityToken/CorporateActions/index.ts
Comment (shuffledex) [reply src/api/entities/SecurityToken/CorporateActions/index.ts:126]: 👍🏻

File: src/api/procedures/modifyCorporateActionsAgent.ts
Comment (shuffledex) [reply src/api/procedures/modifyCorporateActionsAgent.ts:56]: fixed in another PR

File: src/api/procedures/modifyPrimaryIssuanceAgent.ts
Comment (shuffledex) [reply src/api/procedures/modifyPrimaryIssuanceAgent.ts:61]: fixed in another PR

File: src/api/entities/SecurityToken/CorporateActions/index.ts
Comment (VictorVicente) [reply src/api/entities/SecurityToken/CorporateActions/index.ts:126]: I think it is possible, waiting for core answer yet to confirm. 
But I think it is possible to set a `none` AgentGroup to an agent. It is still an agent but has no permissions over the asset. 
I can't imagine the use case for it, but these checks are ok from our side, as it is possible to get empty groups here.

Comment (VictorVicente) [reply src/api/entities/SecurityToken/CorporateActions/index.ts:126]: I think I know the use case. Once an Identity is an Agent, even if they have no AgentGroup, you don't need to invite them again and to wait until they accept to give them an AgentGroup. 
So it could be a way to temporally remove permissions to agents.

File: src/api/entities/Identity/index.ts
Comment (monitz87) [inline src/api/entities/Identity/index.ts:149]: Here you were supposed to change `map(...).includes(...)` with `find`

**Review by monitz87 — CHANGES_REQUESTED** [review]
There are some linting errors (imports that aren't being used)

**Review by monitz87 — APPROVED** [review]

**Pull request merged**
